src/components/EnrolledLearnersForInactiveCoursesTable/data/hooks/useCourseUsers.js

```diff
@@ -0,0 +1,95 @@
+import {
+  useCallback, useMemo, useRef, useState,
+} from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import debounce from 'lodash.debounce';
+import { logError } from '@edx/frontend-platform/logging';
+import EnterpriseDataApiService from '../../../../data/services/EnterpriseDataApiService';
+import EVENT_NAMES from '../../../../eventTracking';
+import { trackDataTableEvent } from '../../../LearnerActivityTable/data/utils';
+
+const applySortByToOptions = (sortBy, apiFieldsForColumnAccessor, options) => {
+  if (!sortBy || sortBy.length === 0) {
+    return;
+  }
+  const orderingStrings = sortBy.map(({ id, desc }) => {
+    const fieldForColumnAccessor = apiFieldsForColumnAccessor[id];
+    if (!fieldForColumnAccessor) {
+      return undefined;
+    }
+    const apiFieldKey = fieldForColumnAccessor.key;
+    return desc ? `-${apiFieldKey}` : apiFieldKey;
+  }).filter(orderingString => !!orderingString);
+  Object.assign(options, {
+    ordering: orderingStrings.join(','),
+  });
+};
+
+const useCourseUsers = (enterpriseId, tableId, apiFieldsForColumnAccessor) => {
+  const shouldTrackFetchEvents = useRef(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [courseUsers, setCourseUsers] = useState({
+    itemCount: 0,
+    pageCount: 0,
+    results: [],
+  });
+
+  const fetchCourseUsers = useCallback(async (args) => {
+    try {
+      setIsLoading(true);
+      const options = {
+        page: args.pageIndex + 1,
+        pageSize: args.pageSize,
+      };
+      applySortByToOptions(args.sortBy, apiFieldsForColumnAccessor, options);
+
+      const response = await EnterpriseDataApiService.fetchEnrolledLearnersForInactiveCourses(enterpriseId, options);
+      const data = camelCaseObject(response.data);
```

Comment:
We only have difference of this endpoint between useCoursesUsers and useCourseEnrollments. So I think we can merge these two hooks into a single one by adding an extra parameter of the endpoint. This will reduce the code and put common functionality at one place.